### PR TITLE
Hotfix/test-provider

### LINKED ISSUE #3: Replace `pnpx`
@phoward38 pretty sure `pnpx` is deprecated and the new method is via the [`pnpm dlx`](https://pnpm.io/cli/dlx) command:

```bash
pnpm dlx degit noclocks noclocks/template-react-ts-starter my-new-project
```

_Originally posted by @jimbrig in https://github.com/noclocks/template-react-ts-starter/pull/1#discussion_r1536541264_

### PR DESCRIPTION
**Pull request by phoward38**
*Have you read the [Contributing Guidelines](https://github.com/noclocks/.github/blob/main/CONTRIBUTING.md)?*

## Changes description

- Adds `postinstall` to require `pnpm`
- Adds `v8` as test `provider` in `vite` config
- Add default logo as `favicon` & update `index.html` so it's shown in `dev` & `preview`
- Additional VSCode settings

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -71,7 +71,7 @@ if you do not have the GitHub CLI installed, you can use the following command:
 
 ```bash
 # assuming project name is: my-new-project
-npx degit noclocks/template-react-ts-starter my-new-project
+pnpx degit noclocks/template-react-ts-starter my-new-project
```
Comment (jimbrig) [inline README.md]: @phoward38 pretty sure `pnpx` is deprecated and the new method is via the [`pnpm dlx`](https://pnpm.io/cli/dlx) command:

```bash
pnpm dlx degit noclocks noclocks/template-react-ts-starter my-new-project
```

**Review by phoward38 — COMMENTED** [review]
Good stuff

File: README.md (hunk 1 above)
Comment (phoward38) [reply README.md]: You're right

File: docs/setup.md [hunk 2]
```diff
@@ -16,7 +16,7 @@ pnpm install --save-dev eslint
 # install airbnb eslint config + peer dependencies
 pnpm install --save-dev eslint-config-airbnb eslint-config-airbnb-typescript eslint-plugin-import eslint-plugin-jsx-a11y eslint-plugin-react eslint-plugin-react-hooks
 # or
-npx install-peerdeps --dev eslint-config-airbnb
+pnpx install-peerdeps --dev eslint-config-airbnb
```
Comment (jimbrig) [inline docs/setup.md]: `pnpm dlx install-peerdeps --dev eslint-config-airbnb`

File: package.json [hunk 3]
```diff
@@ -5,7 +5,7 @@
   "type": "module",
   "license": "See LICENSE.md",
   "scripts": {
-    "preinstall": "npx only-allow pnpm",
+    "preinstall": "pnpx only-allow pnpm",
```
Comment (jimbrig) [inline package.json]: `pnpm dlx only-allow pnpm`

**Review by jimbrig — CHANGES_REQUESTED** [review]
just need to change `pnpx` and everything else looks good. Don't know what the `v8` coverage provider does for the vite config though.

File: docs/setup.md [hunk 4]
```diff
@@ -16,7 +16,7 @@ pnpm install --save-dev eslint
 # install airbnb eslint config + peer dependencies
 pnpm install --save-dev eslint-config-airbnb eslint-config-airbnb-typescript eslint-plugin-import eslint-plugin-jsx-a11y eslint-plugin-react eslint-plugin-react-hooks
```
Comment (phoward38) [inline docs/setup.md:17]: @jimbrig reasoning for using `airbnb` versions?

Comment (jimbrig) [reply docs/setup.md:17]: I found it to be the best setup and most inline with what our standards are.

Reference:

- https://www.npmjs.com/package/eslint-config-airbnb
- https://github.com/airbnb/javascript 
- https://github.com/airbnb/javascript/tree/master/packages/eslint-config-airbnb

**Review by jimbrig — APPROVED** [review]
good2go

**Pull request merged**